### PR TITLE
connect: mux

### PR DESCRIPTION
**Pull request by wasaga**
We want to maintain a single connection to the Connect server. 
`connect-mux` allows multiple update subscribers to listen to updates arriving via Connect service, sharing the underlying gRPC connection.

### CODE REVIEW

File: connect-mux/service.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/hashicorp/go-multierror"
+	"github.com/rs/zerolog/log"
+
+	"github.com/pomerium/zero-sdk/connect"
+	"github.com/pomerium/zero-sdk/fanout"
+)
+
+// Run starts the updates service, listening for updates from the cloud
```
Comment (calebdoxsey) [inline connect-mux/service.go:19]:
```suggestion
// Start starts the updates service, listening for updates from the cloud
```

File: connect-mux/service.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	}
+	onConnected()
+
+	if err = svc.onConnected(ctx); err != nil {
+		return fmt.Errorf("on connected: %w", err)
+	}
+	defer func() {
+		err = multierror.Append(
+			err,
+			nonRetryableError{svc.onDisconnected(ctx)},
```
Comment (calebdoxsey) [inline connect-mux/service.go:102]: doesn't this mean we always return a nonRetryableError?

**Review by calebdoxsey — APPROVED** [review]

**Comment by wasaga** [comment]
oops accidentally merged before GH refreshed and displayed all the comments; will address in a follow-up.

**Comment by wasaga** [comment]
https://github.com/pomerium/zero-sdk/pull/13

**Pull request merged**